arduino-core/src/cc/arduino/packages/discoverers/NetworkDiscovery.java

```diff
@@ -58,19 +58,35 @@ public NetworkDiscovery() {
 
   @Override
   public List<BoardPort> discovery() {
-    List<BoardPort> ports = clonePortsList();
-    Iterator<BoardPort> iterator = ports.iterator();
-    while (iterator.hasNext()) {
+    List<BoardPort> boardPorts = clonePortsList();
+    Iterator<BoardPort> boardPortIterator = boardPorts.iterator();
+    while (boardPortIterator.hasNext()) {
       try {
-        BoardPort board = iterator.next();
-        if (!NetUtils.isReachable(InetAddress.getByName(board.getAddress()), Integer.parseInt(board.getPrefs().get("port")))) {
-          iterator.remove();
+        BoardPort board = boardPortIterator.next();
+
+        InetAddress inetAddress = InetAddress.getByName(board.getAddress());
+        int broadcastedPort = Integer.valueOf(board.getPrefs().get("port"));
+
+        List<Integer> ports = new LinkedList<Integer>();
+        ports.add(broadcastedPort);
+
+        //dirty code: allows non up to date yuns to be discovered. Newer yuns will broadcast port 22
+        if (broadcastedPort == 80) {
+          ports.add(0, 22);
+        }
+
+        boolean reachable = false;
+        for (Integer port : ports) {
+          reachable = reachable || NetUtils.isReachable(inetAddress, port);
```

Comment:
This loop always check for 80 and 22, it may be rewritten as:

```java
reachable = false;
for (Integer port : ports) {
  if (NetUtils.isReachable(inetAddress, port)) {
    reachable = true;
    break;
  }
}
```

so it stops at the first successful check